diff --git a/train/params/P1000/pnet/onsplit_average_reg_10_tanh_large_testing.py b/train/params/P1000/pnet/onsplit_average_reg_10_tanh_large_testing.py
index 164e8aa..2bda9f6 100644
--- a/train/params/P1000/pnet/onsplit_average_reg_10_tanh_large_testing.py
+++ b/train/params/P1000/pnet/onsplit_average_reg_10_tanh_large_testing.py
@@ -18,7 +18,7 @@ data_base = {'id': 'ALL', 'type': 'prostate_paper',
 data = [data_base]
 
 n_hidden_layers = 5
-base_dropout = 0.5
+base_dropout = 0.95
 wregs = [0.001] * 7
 loss_weights = [2, 7, 20, 54, 148, 400]
 wreg_outcomes = [0.01] * 6
@@ -34,7 +34,7 @@ nn_pathway = {
                 'use_bias': True,
                 'w_reg': wregs,
                 'w_reg_outcomes': wreg_outcomes,
-                'dropout': [base_dropout] + [0.1] * (n_hidden_layers + 1),
+                'dropout': [base_dropout] + [0.95] * (n_hidden_layers + 1),
                 'loss_weights': loss_weights,
                 'optimizer': 'Adam',
                 'activation': 'tanh',
