diff --git a/model/builders/builders_utils.py b/model/builders/builders_utils.py
index 909ba26..882cb38 100644
--- a/model/builders/builders_utils.py
+++ b/model/builders/builders_utils.py
@@ -164,9 +164,7 @@ def get_pnet(inputs, features, genes, n_hidden_layers, direction, activation, ac
     #     decision_outcome = Dense(1, activation='linear', name='o_linear{}'.format(1))(outcome)
 
     # drop2 = Dropout(dropout, name='dropout_{}'.format(0))
-    drop2 = Dropout(dropout[0], name='dropout_{}'.format(0))
 
-    outcome = drop2(outcome, training=dropout_testing)
 
     # testing
     if batch_normal:
@@ -233,8 +231,6 @@ def get_pnet(inputs, features, genes, n_hidden_layers, direction, activation, ac
                 decision_outcome = BatchNormalization()(decision_outcome)
             decision_outcome = Activation(activation=activation_decision, name='o{}'.format(i + 2))(decision_outcome)
             decision_outcomes.append(decision_outcome)
-            drop2 = Dropout(dropout, name='dropout_{}'.format(i + 1))
-            outcome = drop2(outcome, training=dropout_testing)
 
             feature_names['h{}'.format(i)] = names
             # feature_names.append(names)
